_search-plugins/knn/nested-search-knn.md

```diff
@@ -0,0 +1,341 @@
+---
+layout: default
+title: k-NN search with nested fields
+nav_order: 21
+parent: k-NN search
+grand_parent: Search methods
+has_children: false
+has_math: true
+---
+
+# k-NN search with nested fields
+With k-NN search using nested fields, you have the flexibility to store multiple vectors within a single document.
+For instance, if your document comprises various components, you can generate a vector value for each component and store them in a nested field within the document.
+When performing a search with a k value of 2, even if a single document has 2 nearest vectors, the search results will include 2 documents: one with the nearest vector and another with the third nearest vector.
```

Comment:
Can this sentence be rephrased into a more simpler form? Maybe first state the overall high-level idea that the document will be included to the result only once and that is based on the highest score/most similarity of all vectors stored in all its nested fields. Example will follow, something like: index has 2 docs, doc A has two fields with vector A1 and A2, and doc B with vector B1. When we run knn search with k == 2. Most similar vectors are A1, A2, B1, but final result will be [A, B] because each doc id will be unique.